tensorflow/python/data/experimental/kernel_tests/cardinality_test.py

```diff
@@ -134,6 +134,15 @@ def _test_combinations():
        lambda: dataset_ops.Dataset.range(5).filter(lambda _: True).take(2),
        cardinality.UNKNOWN),
       ("Take4", lambda: dataset_ops.Dataset.range(5).repeat().take(2), 2),
+      ("Unbatch1",
+       lambda: dataset_ops.Dataset.range(5).batch(2, drop_remainder=True).unbatch(), 4),
+      ("Unbatch2",
+       lambda: dataset_ops.Dataset.range(5).batch(2, drop_remainder=False).unbatch(), cardinality.UNKNOWN),
+      ("Unbatch3",
+       lambda: dataset_ops.Dataset.range(5).batch(2, drop_remainder=True).filter(lambda _: True).unbatch(),
+       cardinality.UNKNOWN),
+      ("Unbatch4", lambda: dataset_ops.Dataset.range(5).batch(2, drop_remainder=True).repeat().unbatch(),
```

Comment:
Add test with 2 components, where only the second component's batch size is known:
```
lambda: dataset_ops.Dataset.zip(
  dataset_ops.Dataset.range(4).batch(2, drop_remainder=False),
  dataset_ops.Dataset.range(5).batch(2, drop_remainder=True))
```

Reply:
Done.